store/tikv/coprocessor_test.go

```diff
@@ -162,3 +163,181 @@ func (s *testCoprocessorSuite) checkEqual(c *C, copRanges *copRanges, ranges []k
+	// result: [c-d) [e-g) [l-o)
+	// split:  [m-o)
+	// result: [m-o)
+	ranges = &copRanges{first: first, mid: mid}
+	s.testZZZ(c, ranges, desc,
+		zzzCase{
+			coprocessorKeyRange("a", "d"),
+			buildKeyRanges("a", "b", "c", "d", "e", "g", "l", "o"),
+		},
+		zzzCase{
+			coprocessorKeyRange("b", "d"),
+			buildKeyRanges("c", "d", "e", "g", "l", "o"),
+		},
+		zzzCase{
+			coprocessorKeyRange("m", "o"),
+			buildKeyRanges("m", "o"),
+		},
+	)
+
+	// range:  [a-b) [c-d) [e-g) [l-o) [q-t)
+	//
+	// split:  [f-o)
+	// result: [f-g) [l-o) [q-t)
+	// split:  [h-p)
+	// result: [l-o) [q-t)
+	// split:  [r-t)
+	// result: [r-t)
+	ranges = &copRanges{first: first, mid: mid, last: last}
+	s.testZZZ(c, ranges, desc,
+		zzzCase{
+			coprocessorKeyRange("f", "o"),
+			buildKeyRanges("f", "g", "l", "o", "q", "t"),
+		},
+		zzzCase{
+			coprocessorKeyRange("h", "p"),
+			buildKeyRanges("l", "o", "q", "t"),
+		},
+		zzzCase{
+			coprocessorKeyRange("r", "t"),
+			buildKeyRanges("r", "t"),
+		},
+	)
+
+	desc = true
+	// reverse, test the desc direction
+	// range:  [c-d) [e-g) [l-o)
+	//
+	// split:  [c-m)
+	// result: [c-d) [e-g) [l-m)
+	// split:  [d-g)
+	// result: [c-d) [e-g)
+	// split:  [f-g)
+	// result: [c-d) [e-g)
+	ranges = &copRanges{mid: mid}
+	s.testZZZ(c, ranges, desc,
+		zzzCase{
+			coprocessorKeyRange("c", "m"),
+			buildKeyRanges("c", "d", "e", "g", "l", "m"),
+		},
+		zzzCase{
+			coprocessorKeyRange("d", "g"),
+			buildKeyRanges("c", "d", "e", "g"),
+		},
+		zzzCase{
+			coprocessorKeyRange("f", "g"),
+			buildKeyRanges("c", "d", "e", "g"),
+		},
+	)
+
+	// range:  [a-b) [c-d) [e-g) [l-o)
+	//
+	// split:  [a-d)
+	// result: [a-b) [c-d)
+	// split:  [b-d)
+	// result: [a-b) [c-d)
+	// split:  [m-o)
+	// result: [a-b) [c-d) [e-g) [l-o)
+	ranges = &copRanges{first: first, mid: mid}
+	s.testZZZ(c, ranges, desc,
+		zzzCase{
+			coprocessorKeyRange("a", "d"),
+			buildKeyRanges("a", "b", "c", "d"),
+		},
+		zzzCase{
+			coprocessorKeyRange("b", "d"),
+			buildKeyRanges("a", "b", "c", "d"),
+		},
+		zzzCase{
+			coprocessorKeyRange("m", "o"),
+			buildKeyRanges("a", "b", "c", "d", "e", "g", "l", "o"),
+		},
+	)
+
+	// range:  [a-b) [c-d) [e-g) [l-o) [q-t)
+	//
+	// split:  [f-o)
+	// result: [a-b) [c-d) [e-g) [l-o)
+	// split:  [h-p)
+	// result: [a-b) [c-d) [e-g) [l-o)
+	// split:  [r-t)
+	// result: [a-b) [c-d) [e-g) [l-o) [q-t)
+	ranges = &copRanges{first: first, mid: mid, last: last}
+	s.testZZZ(c, ranges, desc,
+		zzzCase{
+			coprocessorKeyRange("f", "o"),
+			buildKeyRanges("a", "b", "c", "d", "e", "g", "l", "o"),
+		},
+		zzzCase{
+			coprocessorKeyRange("h", "p"),
+			buildKeyRanges("a", "b", "c", "d", "e", "g", "l", "o"),
+		},
+		zzzCase{
+			coprocessorKeyRange("r", "t"),
+			buildKeyRanges("a", "b", "c", "d", "e", "g", "l", "o", "q", "t"),
+		},
+	)
+}
+
+func coprocessorKeyRange(start, end string) *coprocessor.KeyRange {
+	return &coprocessor.KeyRange{
+		Start: []byte(start),
+		End:   []byte(end),
+	}
+}
+
+type zzzCase struct {
```

Comment:
What does `zzz` means?

Reply:
nothing...any suggestion?

Reply:
Since we added `split` method, then we can just test `split`.

Reply:
Good idea. I'll change it.

Reply:
Done.